src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java

```diff
@@ -132,36 +133,43 @@ public boolean canResolveRequest(@NotNull final IRequestManager manager, final I
      */
     public boolean canResolveForBuilding(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request, @NotNull final AbstractBuilding building)
     {
-        if (building.getBuildingLevel() <= 0 || building.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry() == jobEntry).getAssignedCitizen().isEmpty())
+        if (building.getBuildingLevel() <= 0
+              || !building.hasModule(WorkerBuildingModule.class)
+              || building.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry() == jobEntry).getAssignedCitizen().isEmpty())
         {
             return false;
         }
 
-        if (createsCraftingCycle(manager, request, request))
+        if (createsCraftingCycle(manager, request, request.getRequest(), request))
         {
             return false;
         }
 
-        // As long as we're not resolving food, fast resolve
-        if(!(request.getRequest() instanceof Food))
+        final boolean isFood = !(request.getRequest() instanceof Food);
```

Comment:
`isFood == true` means it's **not** a food request?

Reply:
Good catch